### DISCUSSION TITLE
Option to delete all items of a collection

### DISCUSSION THREAD

**Discussion by Quving** [directus/directus#15818] [Archive]
I've put a large dataset  (350k items) into directus. Now I would like to flush them all but still want to keep the datamodel. Is there any other option instead of delete it page for page which can cover only 1k elements?

**Comment by azrikahar** [comment]
You can do this via the API with a DELETE request to `/items/<your-collection>` (eg. `http://example.com/items/articles`) with the following body:

```json
{
  "query": {
    "limit": -1
  }
}
```

which would delete everything as there's no `filter` query scope it down.

This approach is also shared over here https://github.com/directus/directus/discussions/15320#discussioncomment-3509633 to be used with Delete Data operation within Flows.

**Reply by Quving** [reply]
Thank you!
